src/cargo/util/lints.rs

```diff
@@ -665,13 +665,20 @@ pub fn unexpected_target_cfgs(
         return Ok(());
     };
 
-    if !global_check_cfg.exhaustive {
+    // If we have extra `--check-cfg` args comming from the lints config, we need to
+    // refetch the `--print=check-cfg` with those extra args.
+    let lint_rustflags = pkg.manifest().lint_rustflags();
+    let check_cfg = if lint_rustflags.iter().any(|a| a == "--check-cfg") {
+        Some(target_info.check_cfg_with_extra_args(gctx, &rustc, lint_rustflags)?)
```

Comment:
Why are we calling rustc for this rather than processing them ourselves?

Reply:
Except instead we have a parser for something not quite like `check-cfg` and have to process that.

Reply:
`--check-cfg` and `--print=check-cfg` are different, `--check-cfg` is an input that accepts multiple ways for doing the same thing, it can evolve, and new way of doing things, for example taking one day the [shorthand `target(...)`](https://rust-lang.github.io/rfcs/3239-cfg-target.html), on the other-hand the output of `--print=check-cfg` is meant to be stable, no matter the inputs given in `--check-cfg`.

With `--print=check-cfg` each line a expected key-value (except for `any()=any()` or `any()` but those are exceptions), it's much simpler to parse and it doesn't vary if the input changes, like if introduced the [shorthand `target(...)`](https://rust-lang.github.io/rfcs/3239-cfg-target.html) to `--check-cfg`, the output would stay with the full form, no matter what.